src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -425,6 +426,35 @@ private static int getTokenType(ParseTree node) {
         return tokenType;
     }
 
+    /**
+     * Newly added HTML tag names' tokens have to be introduced before the
+     * {@link JavadocTokenTypes#HTML_TAG_NAME} token in <tt>JavadocLexer</tt>. As a result the
+     * value that ANTLR autromatically generates for <tt>HTML_TAG_NAME</tt> token type changes.
+     * On the other hand checkstyle needs to provide consistent token values to the users in
+     * {@link JavadocTokenTypes} to avoid loss of compatability with the older versions.
+     * Accordingly the token value for <tt>HTML_TAG_NAME</tt> and one other token
+     * ({@link JavadocTokenTypes#EMBED_HTML_TAG_NAME}) are hardcoded and swapped in
+     * <tt>JavadocTokenTypes</tt>.
+     *
+     * @param antlrTokenType value of token in ANTLR
+     * @return value of token as per {@link JavadocTokenTypes}
+     * @see "https://github.com/checkstyle/checkstyle/pull/5455"
+     * @see "https://github.com/checkstyle/checkstyle/issues/5139"
+     */
+    private static int getJavadocTokenType(int antlrTokenType) {
+        final int javadocTokenType;
+        if (antlrTokenType == JavadocTokenTypes.HTML_TAG_NAME) {
+            javadocTokenType = JavadocTokenTypes.EMBED_HTML_TAG_NAME;
```

Comment:
methid name should be named as mapParserTokenToJavadocTokenType
if should be smth like:
```
        if (antlrTokenType == JavadocParser.HTML_TAG_NAME) {
            javadocTokenType = JavadocTokenTypes.HTML_TAG_NAME;
```

Reply:
@romani Sorry for the long delay. Renamed to `mapAntlrTokenTypeToJavadocTokenType`.
Done.